### DISCUSSION TITLE
How to use Jupyter Notebook from VSCode

### DISCUSSION THREAD

**Discussion by adbdkb** [microsoft/vscode-jupyter#13320] [Questions and Answers]
I am new to both Jupyter and Python ( and also VSCode ).  I am trying to get the Jupyter notebook run from within VSCode.  

My VSCode version is 1.77.3 and Jupyter extension version is    v2023.3.1201040234 on Windows 10

I am on a VDI behind a proxy

When I start the notebook from commandline and use it in a browser, it works.  

I tried two set-ups, 
1. Create an .ipynb file and added a cell - print('Hi!')
2. Selected Glue Pyspark as kernel and tried to execute the cell
3. Get the below error

`Glue PySpark - External Jupyter notebook not started

warn 04:37:54.825: Failed to get activated env vars for h:\Software\Python3.9\python in 328ms
> h:\Software\Python3.9\python -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
warn 04:37:55.362: Failed to get activated env vars for h:\Software\Python3.9\python in 564ms
> H:\Software\Python3.9\python -m aws_glue_interactive_sessions_kernel.glue_pyspark.GlueKernel -f ~\AppData\Roaming\jupyter\runtime\kernel-v2-169602ACnowRT2ybo.json
cwd: h:\AWS\Glue\Jupyter-Notebooks
error 04:38:55.556: waitUntilUsed timed out [Error: timeout
	at Timeout.<anonymous> (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\151.node.js:1:17183)
	at listOnTimeout (node:internal/timers:559:17)
	at process.processTimers (node:internal/timers:502:7)]
error 04:38:55.557: Disposing kernel process due to an error [Jp [Error]: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
	at c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59931
	at runNextTicks (node:internal/process/task_queues:61:5)
	at listOnTimeout (node:internal/timers:528:9)
	at process.processTimers (node:internal/timers:502:7)
	at async th.launch (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59973)] {
  category: 'timeout',
  kernelConnectionMetadata: ca {
    kind: 'startUsingLocalKernelSpec',
    kernelSpec: cs {
      specFile: 'c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json',
      interpreterPath: undefined,
      isRegisteredByVSC: undefined,
      name: 'undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel',
      argv: [Array],
      language: 'python',
      executable: 'H:\\Software\\Python3.9\\python',
      display_name: 'Glue PySpark',
      metadata: [Object],
      env: undefined,
      interrupt_mode: undefined
    },
    interpreter: {
      id: 'H:\\SOFTWARE\\PYTHON3.9\\PYTHON',
      sysPrefix: 'h:\\Software\\Python3.9',
      envPath: undefined,
      displayPath: [E],
      envName: '',
      uri: [E],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel'
  }
}
error 04:38:55.558: 
error 04:38:56.157: Kernel Launcher. launch, Class name = ah, completed in 61660ms, has a falsy return value, Arg 1: {"id":".undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel","kernelSpec":{"specFile":"c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json","name":"undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel","argv":["H:\\Software\\Python3.9\\python","-m","aws_glue_interactive_sessions_kernel.glue_pyspark.GlueKernel","-f","{connection_file}"],"language":"python","executable":"H:\\Software\\Python3.9\\python","display_name":"Glue PySpark","metadata":{"vscode":{"originalSpecFile":"c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json","originalDisplayName":"Glue PySpark"}}},"interpreter":{"id":"H:\\SOFTWARE\\PYTHON3.9\\PYTHON","sysPrefix":"h:\\Software\\Python3.9","displayPath":"h:\\Software\\Python3.9\\python","envName":"","displayName":"","envType":"Unknown","isCondaEnvWithoutPython":false,"version":{"major":3,"minor":9,"patch":7,"raw":"3.9.7 (tags/v3.9.7:1016ef3, Aug 30 2021, 20:19:38) [MSC v.1929 64 bit (AMD64)]"},"path":"h:\\Software\\Python3.9\\python"},"kind":"startUsingLocalKernelSpec"}, Arg 2: 60000, Arg 3: "/h:/AWS/Glue/Jupyter-Notebooks/jupyter-vscode-test.ipynb", Arg 4: "h:\\AWS\\Glue\\Jupyter-Notebooks", Arg 5: <argument cannot be serialized for logging> [Jp [Error]: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
	at c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59931
	at runNextTicks (node:internal/process/task_queues:61:5)
	at listOnTimeout (node:internal/timers:528:9)
	at process.processTimers (node:internal/timers:502:7)
	at async th.launch (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59973)] {
  category: 'timeout',
  kernelConnectionMetadata: ca {
    kind: 'startUsingLocalKernelSpec',
    kernelSpec: cs {
      specFile: 'c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json',
      interpreterPath: undefined,
      isRegisteredByVSC: undefined,
      name: 'undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel',
      argv: [Array],
      language: 'python',
      executable: 'H:\\Software\\Python3.9\\python',
      display_name: 'Glue PySpark',
      metadata: [Object],
      env: undefined,
      interrupt_mode: undefined
    },
    interpreter: {
      id: 'H:\\SOFTWARE\\PYTHON3.9\\PYTHON',
      sysPrefix: 'h:\\Software\\Python3.9',
      envPath: undefined,
      displayPath: [E],
      envName: '',
      uri: [E],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel'
  }
}
error 04:38:56.158: Failed to connect raw kernel session: Error: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
warn 04:38:56.159: Error occurred while trying to start the kernel, options.disableUI=false [Jp [Error]: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
	at c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59931
	at runNextTicks (node:internal/process/task_queues:61:5)
	at listOnTimeout (node:internal/timers:528:9)
	at process.processTimers (node:internal/timers:502:7)
	at async th.launch (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59973)] {
  category: 'timeout',
  kernelConnectionMetadata: ca {
    kind: 'startUsingLocalKernelSpec',
    kernelSpec: cs {
      specFile: 'c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json',
      interpreterPath: undefined,
      isRegisteredByVSC: undefined,
      name: 'undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel',
      argv: [Array],
      language: 'python',
      executable: 'H:\\Software\\Python3.9\\python',
      display_name: 'Glue PySpark',
      metadata: [Object],
      env: undefined,
      interrupt_mode: undefined
    },
    interpreter: {
      id: 'H:\\SOFTWARE\\PYTHON3.9\\PYTHON',
      sysPrefix: 'h:\\Software\\Python3.9',
      envPath: undefined,
      displayPath: [E],
      envName: '',
      uri: [E],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel'
  }
}
warn 04:38:56.159: Kernel Error, context = start [Jp [Error]: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
	at c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59931
	at runNextTicks (node:internal/process/task_queues:61:5)
	at listOnTimeout (node:internal/timers:528:9)
	at process.processTimers (node:internal/timers:502:7)
	at async th.launch (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59973)] {
  category: 'timeout',
  kernelConnectionMetadata: ca {
    kind: 'startUsingLocalKernelSpec',
    kernelSpec: cs {
      specFile: 'c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json',
      interpreterPath: undefined,
      isRegisteredByVSC: undefined,
      name: 'undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel',
      argv: [Array],
      language: 'python',
      executable: 'H:\\Software\\Python3.9\\python',
      display_name: 'Glue PySpark',
      metadata: [Object],
      env: undefined,
      interrupt_mode: undefined
    },
    interpreter: {
      id: 'H:\\SOFTWARE\\PYTHON3.9\\PYTHON',
      sysPrefix: 'h:\\Software\\Python3.9',
      envPath: undefined,
      displayPath: [E],
      envName: '',
      uri: [E],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel'
  }
}
warn 04:38:56.280: Failed to get activated env vars for h:\Software\Python3.9\python in 121ms
> h:\Software\Python3.9\python -c "import ipykernel;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
error 04:39:34.541: Error in execution [Jp [Error]: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
	at c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59931
	at runNextTicks (node:internal/process/task_queues:61:5)
	at listOnTimeout (node:internal/timers:528:9)
	at process.processTimers (node:internal/timers:502:7)
	at async th.launch (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59973)] {
  category: 'timeout',
  kernelConnectionMetadata: ca {
    kind: 'startUsingLocalKernelSpec',
    kernelSpec: cs {
      specFile: 'c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json',
      interpreterPath: undefined,
      isRegisteredByVSC: undefined,
      name: 'undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel',
      argv: [Array],
      language: 'python',
      executable: 'H:\\Software\\Python3.9\\python',
      display_name: 'Glue PySpark',
      metadata: [Object],
      env: undefined,
      interrupt_mode: undefined
    },
    interpreter: {
      id: 'H:\\SOFTWARE\\PYTHON3.9\\PYTHON',
      sysPrefix: 'h:\\Software\\Python3.9',
      envPath: undefined,
      displayPath: [E],
      envName: '',
      uri: [E],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel'
  }
}
error 04:39:34.542: Error in execution (get message for cell) [Jp [Error]: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
	at c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59931
	at runNextTicks (node:internal/process/task_queues:61:5)
	at listOnTimeout (node:internal/timers:528:9)
	at process.processTimers (node:internal/timers:502:7)
	at async th.launch (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59973)] {
  category: 'timeout',
  kernelConnectionMetadata: ca {
    kind: 'startUsingLocalKernelSpec',
    kernelSpec: cs {
      specFile: 'c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json',
      interpreterPath: undefined,
      isRegisteredByVSC: undefined,
      name: 'undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel',
      argv: [Array],
      language: 'python',
      executable: 'H:\\Software\\Python3.9\\python',
      display_name: 'Glue PySpark',
      metadata: [Object],
      env: undefined,
      interrupt_mode: undefined
    },
    interpreter: {
      id: 'H:\\SOFTWARE\\PYTHON3.9\\PYTHON',
      sysPrefix: 'h:\\Software\\Python3.9',
      envPath: undefined,
      displayPath: [E],
      envName: '',
      uri: [E],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel'
  }
}

`
5. When I start the notebook externally and select external server from VSCode, I get this error

`External Jupyter Server started - Selected 'Select existing server'

warn 04:45:31.954: Failed to get activated env vars for h:\Software\Python3.9\python in 563ms
> h:\Software\Python3.9\python -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
warn 04:45:32.291: Failed to get activated env vars for h:\Software\Python3.9\python in 365ms
> H:\Software\Python3.9\python -m aws_glue_interactive_sessions_kernel.glue_pyspark.GlueKernel -f ~\AppData\Roaming\jupyter\runtime\kernel-v2-16960CJsv1iX9PPSc.json
cwd: h:\AWS\Glue\Jupyter-Notebooks
error 04:45:45.291: SessionManager cannot enumerate kernelspecs. Returning a default kernel. Specs {}.
error 04:45:52.307: SessionManager cannot enumerate kernelspecs. Returning a default kernel. Specs {}.
error 04:45:52.690: SessionManager cannot enumerate kernelspecs. Returning a default kernel. Specs {}.
error 04:45:52.973: SessionManager cannot enumerate kernelspecs. Returning a default kernel. Specs {}.
error 04:46:32.491: waitUntilUsed timed out [Error: timeout
	at Timeout.<anonymous> (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\151.node.js:1:17183)
	at listOnTimeout (node:internal/timers:559:17)
	at process.processTimers (node:internal/timers:502:7)]
error 04:46:32.491: Disposing kernel process due to an error [Jp [Error]: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
	at c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59931
	at runNextTicks (node:internal/process/task_queues:61:5)
	at listOnTimeout (node:internal/timers:528:9)
	at process.processTimers (node:internal/timers:502:7)
	at async th.launch (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59973)] {
  category: 'timeout',
  kernelConnectionMetadata: ca {
    kind: 'startUsingLocalKernelSpec',
    kernelSpec: cs {
      specFile: 'c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json',
      interpreterPath: undefined,
      isRegisteredByVSC: undefined,
      name: 'undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel',
      argv: [Array],
      language: 'python',
      executable: 'H:\\Software\\Python3.9\\python',
      display_name: 'Glue PySpark',
      metadata: [Object],
      env: undefined,
      interrupt_mode: undefined
    },
    interpreter: {
      id: 'H:\\SOFTWARE\\PYTHON3.9\\PYTHON',
      sysPrefix: 'h:\\Software\\Python3.9',
      envPath: undefined,
      displayPath: [E],
      envName: '',
      uri: [E],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel'
  }
}
error 04:46:32.492: 
error 04:46:33.149: Kernel Launcher. launch, Class name = ah, completed in 61758ms, has a falsy return value, Arg 1: {"id":".undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel","kernelSpec":{"specFile":"c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json","name":"undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel","argv":["H:\\Software\\Python3.9\\python","-m","aws_glue_interactive_sessions_kernel.glue_pyspark.GlueKernel","-f","{connection_file}"],"language":"python","executable":"H:\\Software\\Python3.9\\python","display_name":"Glue PySpark","metadata":{"vscode":{"originalSpecFile":"c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json","originalDisplayName":"Glue PySpark"}}},"interpreter":{"id":"H:\\SOFTWARE\\PYTHON3.9\\PYTHON","sysPrefix":"h:\\Software\\Python3.9","displayPath":"h:\\Software\\Python3.9\\python","envName":"","displayName":"","envType":"Unknown","isCondaEnvWithoutPython":false,"version":{"major":3,"minor":9,"patch":7,"raw":"3.9.7 (tags/v3.9.7:1016ef3, Aug 30 2021, 20:19:38) [MSC v.1929 64 bit (AMD64)]"},"path":"h:\\Software\\Python3.9\\python"},"kind":"startUsingLocalKernelSpec"}, Arg 2: 60000, Arg 3: "/h:/AWS/Glue/Jupyter-Notebooks/jupyter-vscode-test.ipynb", Arg 4: "h:\\AWS\\Glue\\Jupyter-Notebooks", Arg 5: <argument cannot be serialized for logging> [Jp [Error]: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
	at c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59931
	at runNextTicks (node:internal/process/task_queues:61:5)
	at listOnTimeout (node:internal/timers:528:9)
	at process.processTimers (node:internal/timers:502:7)
	at async th.launch (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59973)] {
  category: 'timeout',
  kernelConnectionMetadata: ca {
    kind: 'startUsingLocalKernelSpec',
    kernelSpec: cs {
      specFile: 'c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json',
      interpreterPath: undefined,
      isRegisteredByVSC: undefined,
      name: 'undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel',
      argv: [Array],
      language: 'python',
      executable: 'H:\\Software\\Python3.9\\python',
      display_name: 'Glue PySpark',
      metadata: [Object],
      env: undefined,
      interrupt_mode: undefined
    },
    interpreter: {
      id: 'H:\\SOFTWARE\\PYTHON3.9\\PYTHON',
      sysPrefix: 'h:\\Software\\Python3.9',
      envPath: undefined,
      displayPath: [E],
      envName: '',
      uri: [E],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel'
  }
}
error 04:46:33.149: Failed to connect raw kernel session: Error: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
warn 04:46:33.150: Error occurred while trying to start the kernel, options.disableUI=false [Jp [Error]: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
	at c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59931
	at runNextTicks (node:internal/process/task_queues:61:5)
	at listOnTimeout (node:internal/timers:528:9)
	at process.processTimers (node:internal/timers:502:7)
	at async th.launch (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59973)] {
  category: 'timeout',
  kernelConnectionMetadata: ca {
    kind: 'startUsingLocalKernelSpec',
    kernelSpec: cs {
      specFile: 'c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json',
      interpreterPath: undefined,
      isRegisteredByVSC: undefined,
      name: 'undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel',
      argv: [Array],
      language: 'python',
      executable: 'H:\\Software\\Python3.9\\python',
      display_name: 'Glue PySpark',
      metadata: [Object],
      env: undefined,
      interrupt_mode: undefined
    },
    interpreter: {
      id: 'H:\\SOFTWARE\\PYTHON3.9\\PYTHON',
      sysPrefix: 'h:\\Software\\Python3.9',
      envPath: undefined,
      displayPath: [E],
      envName: '',
      uri: [E],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel'
  }
}
warn 04:46:33.150: Kernel Error, context = start [Jp [Error]: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
	at c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59931
	at runNextTicks (node:internal/process/task_queues:61:5)
	at listOnTimeout (node:internal/timers:528:9)
	at process.processTimers (node:internal/timers:502:7)
	at async th.launch (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59973)] {
  category: 'timeout',
  kernelConnectionMetadata: ca {
    kind: 'startUsingLocalKernelSpec',
    kernelSpec: cs {
      specFile: 'c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json',
      interpreterPath: undefined,
      isRegisteredByVSC: undefined,
      name: 'undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel',
      argv: [Array],
      language: 'python',
      executable: 'H:\\Software\\Python3.9\\python',
      display_name: 'Glue PySpark',
      metadata: [Object],
      env: undefined,
      interrupt_mode: undefined
    },
    interpreter: {
      id: 'H:\\SOFTWARE\\PYTHON3.9\\PYTHON',
      sysPrefix: 'h:\\Software\\Python3.9',
      envPath: undefined,
      displayPath: [E],
      envName: '',
      uri: [E],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel'
  }
}
warn 04:46:33.334: Failed to get activated env vars for h:\Software\Python3.9\python in 183ms
> h:\Software\Python3.9\python -c "import ipykernel;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
error 04:47:12.688: Error in execution [Jp [Error]: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
	at c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59931
	at runNextTicks (node:internal/process/task_queues:61:5)
	at listOnTimeout (node:internal/timers:528:9)
	at process.processTimers (node:internal/timers:502:7)
	at async th.launch (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59973)] {
  category: 'timeout',
  kernelConnectionMetadata: ca {
    kind: 'startUsingLocalKernelSpec',
    kernelSpec: cs {
      specFile: 'c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json',
      interpreterPath: undefined,
      isRegisteredByVSC: undefined,
      name: 'undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel',
      argv: [Array],
      language: 'python',
      executable: 'H:\\Software\\Python3.9\\python',
      display_name: 'Glue PySpark',
      metadata: [Object],
      env: undefined,
      interrupt_mode: undefined
    },
    interpreter: {
      id: 'H:\\SOFTWARE\\PYTHON3.9\\PYTHON',
      sysPrefix: 'h:\\Software\\Python3.9',
      envPath: undefined,
      displayPath: [E],
      envName: '',
      uri: [E],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel'
  }
}
error 04:47:12.689: Error in execution (get message for cell) [Jp [Error]: Unable to start Kernel 'Glue PySpark' due to connection timeout. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
	at c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59931
	at runNextTicks (node:internal/process/task_queues:61:5)
	at listOnTimeout (node:internal/timers:528:9)
	at process.processTimers (node:internal/timers:502:7)
	at async th.launch (c:\Users\myuser\.vscode\extensions\ms-toolsai.jupyter-2023.3.1000892223\out\extension.node.js:17:59973)] {
  category: 'timeout',
  kernelConnectionMetadata: ca {
    kind: 'startUsingLocalKernelSpec',
    kernelSpec: cs {
      specFile: 'c:\\ProgramData\\jupyter\\kernels\\glue_pyspark\\kernel.json',
      interpreterPath: undefined,
      isRegisteredByVSC: undefined,
      name: 'undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel',
      argv: [Array],
      language: 'python',
      executable: 'H:\\Software\\Python3.9\\python',
      display_name: 'Glue PySpark',
      metadata: [Object],
      env: undefined,
      interrupt_mode: undefined
    },
    interpreter: {
      id: 'H:\\SOFTWARE\\PYTHON3.9\\PYTHON',
      sysPrefix: 'h:\\Software\\Python3.9',
      envPath: undefined,
      displayPath: [E],
      envName: '',
      uri: [E],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.undefined.aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel.h:\\Software\\Python3.9\\python.h:\\Software\\Python3.9\\python.-m#aws_glue_interactive_sessions_kernel.glue_pyspark.gluekernel'
  }
}

`

How should I do my set up to get it to work?  i tried to do the steps in couple of other issues here, but they did not work for me

Thank you

[Glue PySpark - External Jupyter notebook not started.txt](https://github.com/microsoft/vscode-jupyter/files/11220495/Glue.PySpark.-.External.Jupyter.notebook.not.started.txt)
 

[External Jupyter Server started - Another Kernel - Existing Jupyter Server.txt](https://github.com/microsoft/vscode-jupyter/files/11220493/External.Jupyter.Server.started.-.Another.Kernel.-.Existing.Jupyter.Server.txt)

[Glue PySpark - External Jupyter notebook not started.txt](https://github.com/microsoft/vscode-jupyter/files/11220494/Glue.PySpark.-.External.Jupyter.notebook.not.started.txt)

**Comment by kvten** [comment]
Have you added python to PATH?

Have you installed all of the following?
- https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter
- https://marketplace.visualstudio.com/items?itemName=ms-python.python

You can try to run code from a python script (.py file).
1. Ctrl+N in VScode to create new page
1. type and run a line of simple python code (Ex: `("Hello World!")`)
